### ISSUE TITLE
Permissions are wrong for /tmp (due to MariaDB?)

### ISSUE THREAD

**Issue by ikari7789** [puphpet/puphpet#1805]
It seems something with the MariaDB install (possibly MySQL install as well since a lot of that code is overlapping) is changing the permissions for the /tmp folder from 1777 to 0775 and changing the owner from root:root to mysql:root.

**Comment by ikari7789** [comment]
I think it's due to this section right here:
https://github.com/puphpet/puphpet/blob/master/archive/puphpet/puppet/nodes/MariaDb.pp#L77-L87

It doesn't exist in the Mysql.pp file.
Perhaps previously the tmpdir was /tmp/mysql or something. I'm unsure. This could probably be fixed by only setting permissions if the tmpdir doesn't equal /tmp.

**Comment by sylvainfilteau** [comment]
It is a problem in my case because Sencha Cmd installer tries to create a file in /tmp and when he can do it, it fails silently. 

As a workaround, I chmod back to 1777 before installing in a exec-once script.

**Comment by vincenzo** [comment]
It's a problem for many, to be honest. Permissions for `/tmp/` should never be different than `1777` and certainly not the `0755` that is being set in the place @ikari7789 has pointed out. I've applied a patch to my puphept-generated files at the moment that completely removes this section: 

```pp

# prevent problems with being unable to create dir in /tmp
  if ! defined(File[$override_options['mysqld']['tmpdir']]) {
    file { $override_options['mysqld']['tmpdir']:
      ensure  => directory,
      owner   => $mariadb_user,
      group   => $mysql::params::root_group,
      mode    => '0775',
      require => Class['mysql::client'],
      notify  => Service[$settings['service_name']]
    }
  }
``` 

But it's annoying as every time I regenerate the files with puphpet, I have to reapply the patch. 

@jtreminio Any chance of this being looked into?

Thanks all.

**Comment by ikari7789** [comment]
This was solved in #1928.
